core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2597,6 +2601,28 @@ public void setAttribute(AlluxioURI path, SetAttributeOptions options)
   }
 
   /**
+   * Checks whether the owner belongs to the group.
+   * @param owner The owner to check
+   * @param group The group to check
+   * @throws FailedPreconditionException if owner does not belong to group
+   * @throws AccessControlException if this operation is not permitted
+   */
+  private void checkUserBelongsToGroup(String owner, String group)
+  throws FailedPreconditionException, AccessControlException {
+    try {
+      List<String> groups = null;
+      groups = CommonUtils.getGroups(owner);
+      if (groups == null || !groups.contains(group)) {
+        throw new FailedPreconditionException("setOwner: owner " + owner
+            + " does not belong to the group " + group);
+      }
+    } catch (IOException e) {
```

Comment:
Since `checkUserBelongsToGroup` is a private utility method which might be used for other operation as well (not for the `setOwner`), is it more generic to just keep all the possible exceptions in this method (with the error message `user not belong to the group...` ? and then you can catch `IOException` in caller sites with more specific error messages associated with the operation type.